utils/aws_api.py

```diff
@@ -0,0 +1,369 @@
+            self.resources[account]['rds_no_owner'] = unfiltered_instances
+
+    def get_resources_without_owner(self, account, resources):
+        resources_without_owner = []
+        for r in resources:
+            if self.has_owner(account, r):
+                continue
+            resources_without_owner.append(r)
+        return resources_without_owner
+
+    def has_owner(self, account, resource):
+        has_owner = False
+        for u in self.users[account]:
+            if resource.lower().startswith(u.lower()):
+                has_owner = True
+                break
+            if '://' in resource:
+                if resource.split('/')[-1].startswith(u.lower()):
+                    has_owner = True
+                    break
+        return has_owner
+
+    def custom_s3_filter(self, account, s3, buckets):
+        type = 's3 bucket'
+        unfiltered_buckets = []
+        for b in buckets:
+            if self.resource_has_special_name(account, type, b):
+                continue
+            # ignore buckets with special tags
+            try:
+                tags = s3.get_bucket_tagging(Bucket=b)['TagSet']
+                if self.resource_has_special_tags(account, type, b, tags):
+                    continue
+                unfiltered_buckets.append(b)
+            except Exception as e:
+                if 'NoSuchTagSet' in e.message:
+                    unfiltered_buckets.append(b)
+        return unfiltered_buckets
+
+    def custom_sqs_filter(self, account, sqs, queues):
+        type = 'sqs queue'
+        unfiltered_queues = []
+        for q in queues:
+            if self.resource_has_special_name(account, type, q):
+                continue
+            # ignore queues with special tags
+            try:
+                tags = sqs.list_queue_tags(QueueUrl=q)['Tags']
+                if self.resource_has_special_tags(account, type, q, tags):
+                    continue
+                unfiltered_queues.append(q)
+            except KeyError as e:
+                if 'Tags' in e.message:
+                    unfiltered_queues.append(q)
+        return unfiltered_queues
+
+    def custom_dynamodb_filter(self, account, session, dynamodb, tables):
+        type = 'dynamodb table'
+        dynamodb_resource = session.resource('dynamodb')
+        unfiltered_tables = []
+        for t in tables:
+            if self.resource_has_special_name(account, type, t):
+                continue
+            # ignore queues with special tags
+            try:
+                table_arn = dynamodb_resource.Table(t).table_arn
+                tags = dynamodb.list_tags_of_resource(
+                    ResourceArn=table_arn)['Tags']
+                if self.resource_has_special_tags(account, type, t, tags):
+                    continue
+                unfiltered_tables.append(t)
+            except KeyError as e:
+                if 'Tags' in e.message:
```

Comment:
this seems a bit brittle, can this be improved perhaps?

Reply:
improved!